src/k8s/pkg/docgen/gomod.go

```diff
@@ -0,0 +1,125 @@
+package docgen
+
+import (
+	"fmt"
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/module"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+func getGoDepModulePath(name string, version string) (string, error) {
+	cachePath := os.Getenv("GOMODCACHE")
+	if cachePath == "" {
+		goPath := os.Getenv("GOPATH")
+		if goPath == "" {
+			goPath = path.Join(os.Getenv("HOME"), "/go")
+		}
+		cachePath = path.Join(goPath, "pkg", "mod")
+	}
+
+	escapedPath, err := module.EscapePath(name)
+	if err != nil {
+		return "", fmt.Errorf(
+			"couldn't escape module path: %s %v", name, err)
+	}
+
+	escapedVersion, err := module.EscapeVersion(version)
+	if err != nil {
+		return "", fmt.Errorf(
+			"couldn't escape module version: %s %v", version, err)
+	}
+
+	path := path.Join(cachePath, escapedPath+"@"+escapedVersion)
+
+	// Validate the path.
+	if _, err := os.Stat(path); err != nil {
+		return "", fmt.Errorf(
+			"Go module path not accessible: %s %s %s. Error: %v.",
+			name, version, path, err)
+	}
+
+	return path, nil
+}
+
+func getDependencyVersionFromGoMod(goModPath string, packageName string, directOnly bool) (string, string, error) {
+	goModContents, err := os.ReadFile(goModPath)
+	if err != nil {
+		return "", "", fmt.Errorf("could not read go.mod file %s. Error: ", goModPath, err)
+	}
+	goModFile, err := modfile.ParseLax(goModPath, goModContents, nil)
+	if err != nil {
+		return "", "", fmt.Errorf("could not parse go.mod file %s. Error: ", goModPath, err)
+	}
+
+	for _, dep := range goModFile.Require {
+		if directOnly && dep.Indirect {
+			continue
+		}
+		if strings.HasPrefix(packageName, dep.Mod.Path) {
+			return dep.Mod.Path, dep.Mod.Version, nil
+		}
+	}
+
+	return "", "", fmt.Errorf("could not find dependency %s in %s", packageName, goModPath)
+}
+
+// getProjectDir retrieves the full path of k8s-snap/src/k8s.
+// For simplicity, we assume that the executable is placed in
+// k8s-snap/src/k8s/bin/(static|dynamic).
+// This will mostly be used to generate the project documentation using
+// "make go.doc".
+func getProjectDir() (string, error) {
```

Comment:
This introduces non-obvious dependencies on the project structure.
Does it make sense to pass the dir e.g. via flag or env?

Reply:
I went with a CLI parameter.